pkg/logql/rangemapper_test.go

```diff
@@ -1750,4 +1751,17 @@ func Test_FailQuery(t *testing.T) {
 	require.Error(t, err)
 	_, _, err = rvm.Parse(`topk(0, sum(count_over_time({app="foo"} | json |  __error__="" [15m])))`)
 	require.Error(t, err)
+	// Check fixes for bug where missing or empty parameters for regexp and pattern parsers threw a panic
+	// Missing parameter to regexp parser
+	_, _, err = rvm.Parse(`topk(10,sum by(namespace)(count_over_time({application="nginx", site!="eu-west-1-dev"} |= "/artifactory/" != "api" != "binarystore" | regexp [1d])))`)
+	require.ErrorIs(t, err, logqlmodel.ErrParse)
+	// Empty parameter to regexp parser
+	_, _, err = rvm.Parse(`topk(10,sum by(namespace)(count_over_time({application="nginx", site!="eu-west-1-dev"} |= "/artifactory/" != "api" != "binarystore" | regexp ` + "``" + ` [1d])))`)
```

Comment:
Does it also fail if we use other empty string with for instance `""` or `''`?

Reply:
And shouldn't we also add equivalent unit tests in `pkg/logql/syntax/ast_test.go`? 🤔

Reply:
Yes, it also fails with a panic with the other empty strings (I just ran the test again without the fixes).
Additional unit test were added as well.